ambari-web/app/controllers/wizard/step8_controller.js

```diff
@@ -1467,16 +1462,25 @@ App.WizardStep8Controller = App.WizardStepController.extend(App.AddSecurityConfi
    * @param {String} componentName
    * @method registerHostsToComponent
    */
-  registerHostsToComponent: function (hostNames, componentName, serviceName) {
+  registerHostsToComponent: function (hostNames, componentName) {
     if (!hostNames.length) return;
 
+    let serviceName,
+        serviceGroupName;
     var queryStr = '';
     hostNames.forEach(function (hostName) {
       queryStr += 'Hosts/host_name=' + hostName + '|';
     });
     //slice off last symbol '|'
     queryStr = queryStr.slice(0, -1);
 
+    this.get('selectedServices').forEach( function (service) {
+      if (service.get('serviceComponents').findProperty('componentName', componentName)) {
```

Comment:
Do we see a potential issue if a two mpacks have the same serviceComponentName?
@swapanshridhar

Reply:
@swapanshridhar @mradha25 The issue will be caused if a service is mapped to multiple mpacks. Also two services cannot have same component names right?

Reply:
@swapanshridhar @mradha25 ping!

Reply:
@mradha25 @swapanshridhar merging this. Let me know incase any issues arise after this commit.